### PR TITLE
Consistent set/get interface for system prompt and turns

### PR DESCRIPTION
**Pull request by hadley**
@cpsievert do you want to take a look at this too?

### CODE REVIEW

File: R/chat.R [hunk 1]
```diff
@@ -34,11 +34,11 @@ Chat <- R6::R6Class("Chat",
       private$echo <- echo
     },
 
-    #' @description The turns that have been sent and received so far
+    #' @description Retrive the turns that have been sent and received so far
```
Comment (cpsievert) [inline R/chat.R:37]:
```suggestion
    #' @description Retrieve the turns that have been sent and received so far
```

File: R/chat.R [hunk 2]
```diff
@@ -50,6 +50,37 @@ Chat <- R6::R6Class("Chat",
       }
     },
 
+    #' @description Replace existing turns with a new list.
+    #' @param value A list of [Turn]s.
+    set_turns = function(value) {
+      private$.turns <- normalize_turns(value, self$system_prompt)
```
[the remark this answers is not in the document]
Comment (cpsievert) [reply R/chat.R:56]: Also, if `value` contains a `Turn("system", ...)`, I think we'd end up with multiple system prompts, which seems problematic?

FWIW, the approach I was taking here was to throw if `value` contains a `Turn("system", ...)`

[the remark this answers is not in the document]
Comment (hadley) [reply R/chat.R:56]: Oops. I wonder why my tests didn't catch this?

**Comment by jcheng5** [comment]
API looks fine to me, I don't have much of an opinion here.

File: R/chat.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (hadley) [reply R/chat.R]: Thanks for catching this — I had a bit of a miss on the logic here.

**Pull request merged**